src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -142,6 +149,32 @@ bool KeyboardManagerState::AddSingleKeyRemap(const DWORD& originalKey, const DWO
     return true;
 }
 
+// Function to add a new App specific shortcut remapping
+bool KeyboardManagerState::AddAppSpecificShortcut(const std::wstring& app, const Shortcut& originalSC, const Shortcut& newSC)
+{
+    std::lock_guard<std::mutex> lock(appSpecificShortcutReMap_mutex);
+
+    // Check if there are any app specific shortcuts for this app
+    auto appIt = appSpecificShortcutReMap.find(app);
+    if (appIt != appSpecificShortcutReMap.end())
+    {
+        // Check if the shortcut is already remapped
+        auto shortcutIt = appSpecificShortcutReMap[app].find(originalSC);

```

Comment:
[clarification] Is this limiting number of shortcuts to a functionality? For example : It is not possible to map both "ctrl+A" and "ctrl+B" to "ctrl+v" ?

Reply:
I think it is just checking the `Original Shortcut`, not the `remapped shortcut`. So in this case, it would be checking if Ctrl+A is already remapped to something else.

Reply:
No, this ensures that you can't remap `Ctrl+A` to `Ctrl+V` and `Ctrl+A` to `Ctrl+C`, since if that was allowed we wouldn't know how which remap should happen.